### PR TITLE
fix(IRegistrationContext): Add missing TaskProcessing methods

### PR DESCRIPTION
**Pull request by provokateurin**
## Summary

Follow up for https://github.com/nextcloud/server/pull/45094. The methods are present in the implementation but not in the interface, so the interface is not complete and also psalm complains about it.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by marcelklehr — APPROVED** [review]
Strange that the CI was green when  I merged.

**Comment by provokateurin** [comment]
Well the server CI is fine because the method is never called here. Just trying to call it from app_api it complains because the method is missing on the interface.

**Review by bigcat88 — APPROVED** [review]

**Review by julien-nc — APPROVED** [review]

File: lib/public/AppFramework/Bootstrap/IRegistrationContext.php [hunk 1]
```diff
@@ -410,4 +410,26 @@ public function registerSetupCheck(string $setupCheckClass): void;
...
+	/**
+	 * Register an implementation of \OCP\TaskProcessing\IProvider that
+	 * will handle the implementation of task processing
+	 *
+	 * @param string $taskProcessingProviderClass
+	 * @psalm-param class-string<\OCP\TaskProcessing\IProvider> $taskProcessingProviderClass
+	 * @return void
+	 * @since 30.0.0
+	 */
+	public function registerTaskProcessingProvider(string $taskProcessingProviderClass): void;
```
Comment (nickvergessen) [inline lib/public/AppFramework/Bootstrap/IRegistrationContext.php:423]: @provokateurin Please document in https://docs.nextcloud.com/server/latest/developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_30.html#id1

Comment (provokateurin) [reply lib/public/AppFramework/Bootstrap/IRegistrationContext.php:423]: The documentation here should also be added by https://github.com/nextcloud/server/pull/45094

**Pull request merged**